### PR TITLE
Rewrite image & color mapping

### PR DESCRIPTION
**Pull request by Playwo**
### This is unfinished!

**Purpose**
Currently the ImgGenBiomeSource class is hard to understand and contains a lot of code. This PR splits it up into multiple reusable classes which should also be able to be used for the structures. (ToDo)

I made one change in functionality: Scaling.
The current scale was pretty weird for me and It was easy to implement it in a different way. 
It seems to work fine but I don't have a great way of testing my code.

I will push the required changes for the structures in a while, but feel free to give this a try and let me know if something is off.

### CODE REVIEW

**Comment by Playwo** [comment]
Can you please provide me with some contact information (preferably discord) 
I have some questions to things I found while going through the code.

**Comment by 28Pollux28** [comment]
The scale part was a bit tricky and done a while ago. I was gonna work on the #4 suggestion and try to review it after, but since you did it, I'll change my order. I'm going to test what you send, and we'll see how it could be done on structures.
Here's my discord : Pollux#5784

**Comment by Playwo** [comment]
#4 should be relatively simple to do by just adding a modulo operation in [here](https://github.com/Playwo/ImageWorldGenerator/blob/master/src/main/java/eu/pollux28/imggen/data/ImageDataProvider.java#L28-L29)
#5 and structures should both work similar to how I implemented [this](https://github.com/Playwo/ImageWorldGenerator/blob/master/src/main/java/eu/pollux28/imggen/data/BiomeColorConverter.java)

File: src/main/java/eu/pollux28/imggen/ImgGen.java [hunk 1]
```diff
@@ -30,19 +30,18 @@
     @Override
     @SuppressWarnings("ResultOfMethodCallIgnored")
     public void onInitialize() {
-        if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT){
-            levelGeneratorType = new ImgGenType("imggen");
-        }
+        //if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT){
+         //   levelGeneratorType = new ImgGenType("imggen");
+        //}
```
Comment (28Pollux28) [inline src/main/java/eu/pollux28/imggen/ImgGen.java:35]: those are necessary to show the generator type on client side

**Pull request merged**